spot_choreo_utils/spot_choreo_utils/choreo_creation/choreo_builders/sequence_builder.py

```diff
@@ -85,6 +92,230 @@ def add_animation(self, animation: Animation, start_time: float) -> None:
+            duration_sec: duration of the move, in seconds. This will repeat the move.
+            roll: target body roll, in radians. Clamped to [-0.5, 0.5]. Default is 0.05.
+            pitch: target body pitch, in radians. Clamped to [-0.5, 0.5]. Default is 0.05.
+            yaw: target body yaw, in radians. Clamped to [-0.5, 0.5]. Default is 0.00.
+            return_to_start_pose: will return body to the previous pose by the end of this move.
+
+        Returns:
+            None
+        """
+
+        # Re-frame from time to slices
+        slices_per_second = self._sequence.slices_per_minute / 60
+        start_slice = int(start_sec * slices_per_second)
+
+        # Calculate the slices to request based on duration
+        requested_slices = max(int(duration_sec * slices_per_second), 1)
+
+        # Sanitize inputs
+        start_sec = max(start_sec, 0)
+        duration_sec = max(start_sec, 0)
+        roll = max(min(roll, 0.5), -0.5)
+        pitch = max(min(pitch, 0.5), -0.5)
+        yaw = max(min(yaw, 0.5), -0.5)
+
+        # Construct the move-specific parameters
+        rotate_body_params = RotateBodyParams()
+        rotate_body_params.EulerZYXValue.roll.value = roll
+        rotate_body_params.EulerZYXValue.pitch.value = pitch
+        rotate_body_params.EulerZYXValue.yaw.value = yaw
+        rotate_body_params.return_to_start_pose.value = return_to_start_pose
+
+        # Set up its role within the sequence
+        move_params = MoveParams()
+        move_params.type = "rotate_body"
+        move_params.start_slice = start_slice
+        move_params.requested_slices = requested_slices
+        move_params.rotate_body_params.CopyFrom(rotate_body_params)
+
+        # Add to the sequence
+        self._sequence.moves.append(move_params)
+
+    def add_sway(
+        self,
+        start_sec: float,
+        duration_sec: float,
+        vertical: float = 0.00,
+        horizontal: float = 0.05,
+        roll: float = 0.00,
+        pivot: Any = Pivot.PIVOT_CENTER,  # since Pivot is a non-single protobuf type, we must use Any
+        style: Any = SwayParams.SwayStyle.SWAY_STYLE_STANDARD,
+        pronounced: float = 0.5,
+        hold_zero_axes: bool = False,
+    ) -> None:
+        """
+        Add a sway to the sequence
+
+        Args:
+            start_sec: absolute time stamp at which point the move should start,
+                in seconds.
+            duration_sec: duration of the move, in seconds. This will repeat the move.
+            vertical: target body vertical displacement, in meters. Clamped to
+                [-0.2, 0.2]. Default is 0.00.
+            horizontal: target body horizontal displacement, in meters. Clamped to
+                [-0.4, 0.4]. Default is 0.05.
+            roll: target body roll, in radians. Clamped to [-0.2, 0.2]. Default is 0.00.
+            pivot: which portion of the body remains stationary. 0 = front, 2 = hind,
+                3 = center. Default is center.
```

Comment:
seems like from the docs that 0 is unknown:
```suggestion
            pivot: which portion of the body remains stationary. 1 = front, 2 = hind,
                3 = center. Default is center.
```
also, it might be more clear if you tell the user to pass in Pivot enums rather than numbers? might be more readable, and if BD ever hypothetically decides to change the enum mapping then this will stay in sync. (i guess same comment for the `style` argument below).

Reply:
That is a good point about passing Pivot enum would protect against BD updating this in the future. I'll do that.

Reply:
Fixed [here](https://github.com/bdaiinstitute/spot_choreo_utils/pull/18/commits/578f575f58c6b7f432691173678ffbdacecb6b79).